### PR TITLE
Backport libcompat cleanups committed upstream

### PR DESCRIPTION
**Pull request by jrtc27**

### CODE REVIEW

File: include/paths.h [hunk 1]
```diff
@@ -37,8 +37,8 @@
 
 #include <sys/cdefs.h>
 
-#ifdef COMPAT_64BIT
-#define	_PATH_LOCALBASE	"/usr/local64"
+#ifdef COMPAT_libcompat
```
Comment (bsdjhb) [inline include/paths.h:40]: Hmmm, this has subtle changes for COMPAT32 that aren't present upstream.  I think we also have decided against having /usr/local64c on hybrid worlds?

File: lib/libutil/getlocalbase.c [hunk 2]
```diff
@@ -38,8 +38,8 @@ __FBSDID("$FreeBSD$");
 #include <unistd.h>
 
 #ifndef _PATH_LOCALBASE
-#ifdef COMPAT_64BIT
-#define	_PATH_LOCALBASE	"/usr/local64"
+#ifdef COMPAT_libcompat
```
Comment (bsdjhb) [inline lib/libutil/getlocalbase.c:41]: Same comment as for LOCALBASE in include/paths.h.

File: include/paths.h (hunk 1 above)
Comment (jrtc27) [reply include/paths.h:40]: It does, but we will at least want a local64cb in time, and I don't really see what harm it does for lib32? As for hybrid, we don't support ports or packages there at all so I don't personally care what happens.

Comment (jrtc27) [reply include/paths.h:40]: But if desired I can just change this to
```
#if defined(COMPAT_LIB64)
#define _PATH_LOCALBASE "/usr/local" COMPAT_libcompat
```
and then later add `|| defined(COMPAT_LIB64CB)`

Comment (bsdjhb) [reply include/paths.h:40]: I think I might prefer we do that.

File: lib/libutil/getlocalbase.c (hunk 2 above)
Comment (bsdjhb) [reply lib/libutil/getlocalbase.c:41]: Also, I wonder why this doesn't use paths.h upstream rather than duplicating the logic.

Comment (jrtc27) [reply lib/libutil/getlocalbase.c:41]: Oh, probably for bootstrapping on older FreeBSD; the paths.h define and this code came in around the same time

Comment (bsdjhb) [reply lib/libutil/getlocalbase.c:41]: Humm, that seems odd indeed.

Comment (jrtc27) [reply lib/libutil/getlocalbase.c:41]: Except it's not a bootstrap library... so who knows

File: include/paths.h (hunk 1 above)
Comment (brooksdavis) [reply include/paths.h:40]: FWIW, I suspect upstream will find they want /usr/local32 at some point after we kill off 32-worlds, but I don't think I'm in a rush to push that through

**Review by bsdjhb — APPROVED** [review]

**Pull request merged**